cheatsheets/Vulnerable_Dependency_Management_Cheat_Sheet.md

```diff
@@ -0,0 +1,233 @@
+
+```java
+public void callFunctionWithRCEIssue(String externalInput){
+    //Apply input validation on the external input using regex
+    if(Pattern.matches("[a-zA-Z0-9]{1,50}", externalInput)){
+        //Call the flawed function using safe input
+        functionWithRCEIssue(externalInput);
+    }else{
+        //Log the detection of exploitation
+        SecurityLogger.warn("Exploitation of the RCE issue XXXXX detected !");
+        //Raise an exception leading to a generic error send to the client...
+    }
+}
+```
+
+If the provider has provided nothing about the vulnerability then apply the [Case 3](#-case-3) skipping the *step 2* of this case. We assume here that, at least, the [CVE](https://en.wikipedia.org/wiki/Common_Vulnerabilities_and_Exposures) has been provided.
+
+**Step 2:**
+
+If the provider has given the exploitation code then try to execute it against the features where protection code has been added in order to verify that the protection code is effective.
+
+If you have a set of automated unit or integration or functionnal or security tests that exists for the application then run them to verify that the protection code added do not imapct the stability of the application.
+
+Add a comment in the project *README* explaining that the issue (specify the related [CVE](https://en.wikipedia.org/wiki/Common_Vulnerabilities_and_Exposures)) is handled during the waiting time of a patched version because the detection tool will continue to raise an alert on this dependency. 
+
+**Note:** You can add the dependency to the ignore list but the ignore scope for this dependency must only cover the [CVE](https://en.wikipedia.org/wiki/Common_Vulnerabilities_and_Exposures) related to the vulnerability because a dependency can be impacted by several vulnerabilities having each one is proper [CVE](https://en.wikipedia.org/wiki/Common_Vulnerabilities_and_Exposures).
+
+## Case 3
+
+### Context
+
+Provider inform the team that he cannot fix the issue so no patched version will be released at all (apply also if provider do not want to fix the issue or do not answer at all).
+
+The only information given to the development team is the [CVE](https://en.wikipedia.org/wiki/Common_Vulnerabilities_and_Exposures).
+
+**Note:** This case is really complex and time consuming and is generally used as last resort.
+
+### Ideal condition of application of the approach
+
+Nothing specific because here we are in *patch yourself* condition.
+
+### Approach
+
+**Step 1:**
+
+If we are in this case due to one of the following condition, it's a good idea to start a parallel study to find another component better maintained or if it's a commercial component with support **then put pressure** on the provider with the help of your [Chief Risk Officer](https://en.wikipedia.org/wiki/Chief_risk_officer) (fallback possible to [Chief Information Security Officer](https://en.wikipedia.org/wiki/Chief_information_security_officer)):
```

Comment:
If this is open source library we development team can create patch and create PR - that way we can protect our company/application very quickly but also helps others.

Reply:
Note added with your feedback in coming commit.